Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -73,74 +93,203 @@ namespace AZ::DocumentPropertyEditor
+                };
+
+                // Find the row that contains the PropertyEditor for our actual container (if it exists)
+                auto visitorFlags =
+                    Dom::PrefixTreeTraversalFlags::ExcludeChildPaths | Dom::PrefixTreeTraversalFlags::TraverseMostToLeastSpecific;
+                impl->m_containers.VisitPath(path, findContainerProcedure, visitorFlags);
+
+                if (containerRow.IsNode())
+                {
+                    // Look within the Row for a PropertyEditor that has a SerializedPath field.
+                    // This will be the container's editor w/ attributes.
+                    for (auto it = containerRow.ArrayBegin(); it != containerRow.ArrayEnd(); ++it)
+                    {
+                        if (it->IsNode() && it->GetNodeName() == GetNodeName<Nodes::PropertyEditor>())
+                        {
+                            auto serializedFieldIt = it->FindMember(Reflection::DescriptorAttributes::SerializedPath);
+                            if (serializedFieldIt != it->MemberEnd())
+                            {
+                                return *it;
+                            }
+                        }
+                    }
+                }
+                return {};
             }
 
             void OnRemoveElement(ReflectionAdapterReflectionImpl* impl, const AZ::Dom::Path& path)
             {
-                AZ_Assert(m_elementInstance != nullptr, "Attempted to remove an element without a defined element instance");
-                m_container->RemoveElement(m_instance, m_elementInstance, impl->m_serializeContext);
+                void* instanceToRemove = m_elementInstance;
+
+                auto associativeContainer = m_container->GetAssociativeContainerInterface();
+                if (associativeContainer)
+                {
+                    const AZ::SerializeContext::ClassElement* containerClassElement =
+                        m_container->GetElement(m_container->GetDefaultElementNameCrc());
+                    instanceToRemove = associativeContainer->GetElementByKey(m_containerInstance, containerClassElement, m_elementInstance);
+
+                }
+
+                m_container->RemoveElement(m_containerInstance, instanceToRemove, impl->m_serializeContext);
 
                 auto containerNode = GetContainerNode(impl, path);
-                Nodes::PropertyEditor::RemoveNotify.InvokeOnDomNode(containerNode, m_elementIndex);
                 Nodes::PropertyEditor::ChangeNotify.InvokeOnDomNode(containerNode);
-
                 impl->m_adapter->NotifyResetDocument();
             }
+
+            AZ::SerializeContext::IDataContainer* m_container = nullptr;
+            void* m_containerInstance = nullptr;
+            void* m_elementInstance = nullptr;
+        };
+
+        struct ContainerEntry
+        {
+            AZStd::unique_ptr<BoundContainer> container;
+            AZStd::unique_ptr<ContainerElement> element;
```

Comment:
Could you use m_ for the members here?
```suggestion
            AZStd::unique_ptr<BoundContainer> m_container;
            AZStd::unique_ptr<ContainerElement> m_element;
```